### DISCUSSION TITLE
unable to run predictions

### DISCUSSION THREAD

**Discussion by mhochst** [talmolab/sleap#1344] [Help!] [answered]
![image](https://github.com/talmolab/sleap/assets/122642039/d42d1378-1735-4121-91fc-f696ac7403ff)
![image](https://github.com/talmolab/sleap/assets/122642039/ec00071a-d11d-4253-a147-6f0822a9ad78)
I recently ran training on sleap on all frames in video when i meant to just have it run through random frames, ive deleted all the non user predicted instances but now i am unable to run training on any set of frames, im not quite sure what went wrong or how to fix this. any advice would be appreciated.

**Comment by roomrys** [comment]
Hi @mhochst,

Sorry for the bit of lag, but the message in your terminal would greatly help with diagnosing your problem if you are able to provide that.

Thanks,
Liezl

**Reply by mhochst** [reply]
i should have included it in the first message, my apologies. im just not sure what broke here and how to identify and fix the problem without starting the dataset over again

![image](https://github.com/talmolab/sleap/assets/122642039/d850854c-0bca-486f-b42b-0329463c3c15)
![image](https://github.com/talmolab/sleap/assets/122642039/8402766e-2ede-463c-8c42-296063120afa)

**Comment by roomrys** [comment, marked as answer]
Hi @mhochst,

Thanks for the messages! The error:
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: Shape of tensor EagerPyFunc [X,X,1] is not compatible with expected shape [X,X,3]
```
looks similar to #922:
>Do you happen to have both color and grayscale videos in your project? While SLEAP should be able to handle training on mixed color/grayscale project, we have an open issue #759 on this at the moment. The current workaround would be to open the GUI, check the number of channels for each video in the Video view, and use the "Toggle Grayscale" button to convert each video to have the same number of channels.
Let us know if that solves the issue.

The more concerning this would be if you did not import any new videos or did not toggle the color on any of the videos?

Thanks,
Liezl

**Reply by mhochst** [reply]
I'll take a look when I get back to my desk, I didn't add any new videos to the project, it was working fine this week til I had it run prediction on every single frame on accident

**Reply by mhochst** [reply]
that worked!!! thank you so much!
